### PR TITLE
Move the results download to the electron app

### LINKED ISSUE #161: Download results
### Description of the requested feature:

When a pipeline is being run, the result files generated by the pipeline should be downloaded to my local machine.

### Brief explanation on why you want/need said feature:

This feature is required for the cloud version of zetaforge, where view images might reference private resources.

### Do you intend on coding this feature yourself?

Yes

### Which zetaforge components would be affected by this new feature:

- [X] client
- [ ] anvil
- [ ] driver

### Share any code snippets related to the feature here(optional):

_No response_

### Provide an API design for the feature(optional):

- During execution, the results are downloaded to the buffer file
- When opening an execution, the result files are downloaded to the buffer file
- The downloaded result files are accessed through the express server

### PR DESCRIPTION
**Pull request by FGRCL**
Closes #161

### CODE REVIEW

File: frontend/server/s3.js [hunk 1]
```diff
@@ -170,3 +171,15 @@ export async function getFile(key, destinationPath, anvilConfiguration) {
...
+export async function syncS3ToLocalDirectory(
+  s3Prefix,
+  localPath,
+  anvilConfiguration,
+) {
+  const client = getClient(anvilConfiguration);
+  const { sync } = new S3SyncClient({ client: client });
+
+  const s3Path = `s3://${anvilConfiguration.s3.bucket}/${s3Prefix}`;
+  await sync(s3Path, localPath, { del: true });
```
Comment (jmagoon) [inline frontend/server/s3.js:184]: If I load a pipeline that no longer exists in my s3 bucket (e.g. happens for me when I stop and restart orbstack, s3 is cleared), will the `del` param here clear my local history?

Comment (FGRCL) [reply frontend/server/s3.js:184]: Not tested, but I'd assume it would yeah

Comment (jmagoon) [reply frontend/server/s3.js:197]: Oh, I'm saying we don't want to do that. I would suggest we don't `del` without a more robust strategy for that.

Comment (FGRCL) [reply frontend/server/s3.js]: Ah ok, that's easy to fix

**Pull request merged**